### PR TITLE
Split addSystem to addEmulator / addSystem

### PR DESCRIPTION
**Pull request by joolswills**
Currently we have just addSystem which is used to add emulators and systems all at the same time. This is fine, except in some cases where we have multiple calls - eg for fuse where we have a 48k and 128k "emulator" (calling fuse with different parameters etc). Right now this causes the frontend system to be updated twice. 

Now we have other frontends, it would be better to just try and add a system once as needed. Also since the addSystem call handled the removal, we would also try and "remove" it multiple times when there was more than one emulator entry for that module.

By splitting it, we can now just add multiple emulators, and then add the system afterwards (it has to be in the order due to when in removal mode - system must be removed last).

As well as avoiding adding/updating a system multiple times it makes the code cleaner as the functions do what the name suggests, rather than one function that does multiple tasks.

Note backward compatibility is included for 3rd party repos like @zerojay 's - but they should be updated soon as it may be removed in the future.

This is quite a large patchset and there may be bugs, so I will give it some more testing before merging - feel free to test.

### CODE REVIEW

**Comment by joolswills** [comment]
Oh, and there is another change to use the platforms.cfg for the arcade "platforms/theme" - this simplifies the calls as we had cases where we were passing "system platform theme" as a single parameter to the old addSystem.

**Comment by joolswills** [comment]
I forgot to move some full name/extensions to the addsystem call so some more work needs doing on this

File: scriptmodules/emulators/oricutron.sh [hunk 1]
```diff
@@ -59,8 +59,9 @@ function configure_oricutron() {
     for machine in atmos oric1 o16k telestrat pravetz; do
         default=0
         [[ "$machine" == "atmos" ]] && default=1
-        addSystem 1 "$md_id-$machine" "oric" "pushd $md_inst; $md_inst/oricutron --machine $machine %ROM% --fullscreen; popd" "Oric 1/Atmos" ".dsk .tap"
+        addEmulator 1 "$md_id-$machine" "oric" "pushd $md_inst; $md_inst/oricutron --machine $machine %ROM% --fullscreen; popd" "Oric 1/Atmos" ".dsk .tap"
```
Comment (HerbFargus) [inline scriptmodules/emulators/oricutron.sh]: needs to be added to the platform.cfg

**Comment by joolswills** [comment]
Yep - I've not finished yet :)

**Comment by joolswills** [comment]
This *should* be ready now - but there may be some bugs. I'm doing some final testing so if you notice any typos mistakes please do let me know. I just fixed a couple of typos, and there may be more.

File: platforms.cfg [hunk 2]
```diff
@@ -67,19 +74,31 @@ gb_fullname="Game Boy"
...
+love_fullname="Love"
+
 intellivision_exts=".int .bin"
 intellivision_fullname="Intellivision"
 
+macintosh_exts=".txt"
+macintosh_fullname="Apple Macintosh"
+
+macintoshplus_exts=".txt .dsk"
+macintosh_fullname="Apple Macintosh Plus"
```
Comment (HerbFargus) [inline platforms.cfg]: macintoshplus_fullname ?

File: scriptmodules/libretrocores/lr-picodrive.sh [hunk 3]
```diff
@@ -42,17 +42,11 @@ function install_lr-picodrive() {
...
-    ensureSystemretroconfig "mastersystem"
-    ensureSystemretroconfig "segacd"
-    ensureSystemretroconfig "sega32x"
-
-    addSystem 1 "$md_id" "mastersystem" "$md_inst/picodrive_libretro.so"
-    addSystem 1 "$md_id" "megadrive" "$md_inst/picodrive_libretro.so"
-    addSystem 1 "$md_id" "segacd" "$md_inst/picodrive_libretro.so"
-    addSystem 1 "$md_id" "sega32x" "$md_inst/picodrive_libretro.so"
+    local system
+    for system in megadrrive mastersystem segacd sega32x; do
```
Comment (HerbFargus) [inline scriptmodules/libretrocores/lr-picodrive.sh]: megadrive (only with one R ;) )

**Pull request merged**